apps/etterlatte-trygdetid/src/main/kotlin/trygdetid/regler/BeregnTrygdetid.kt

```diff
@@ -24,63 +25,76 @@ data class TotalTrygdetidGrunnlag(
     beskrivelse = "Periode fra",
     finnFaktum = TrygdetidPeriodeGrunnlag::periodeFra,
     finnFelt = { it }
 )
 
 val periodeTil: Regel<TrygdetidPeriodeGrunnlag, LocalDate> = finnFaktumIGrunnlag(
-    gjelderFra = TRYGDETID_OMS_DATO,
+    gjelderFra = TRYGDETID_DATO,
     beskrivelse = "Periode til",
     finnFaktum = TrygdetidPeriodeGrunnlag::periodeTil,
     finnFelt = { it }
 )
 
-val beregnTrygdetidMellomToDatoer = RegelMeta(
-    gjelderFra = TRYGDETID_OMS_DATO,
-    beskrivelse = "Beregner trygdetid mellom to datoer i år, måneder og dager",
+val beregnTrygdetidForPeriode = RegelMeta(
+    gjelderFra = TRYGDETID_DATO,
+    beskrivelse = "Beregner trygdetid fra og med periodeFra til og med periodeTil i år, måneder og dager",
     regelReferanse = RegelReferanse(id = "REGEL-TRYGDETID-BEREGNE-PERIODE")
 ) benytter periodeFra og periodeTil med { periodeFra, periodeTil ->
     Period.between(periodeFra, periodeTil.plusDays(1))
 }
 
 val beregnetTrygdetidPerioder: Regel<TotalTrygdetidGrunnlag, List<Period>> = finnFaktumIGrunnlag(
-    gjelderFra = TRYGDETID_OMS_DATO,
+    gjelderFra = TRYGDETID_DATO,
     beskrivelse = "Beregnet trygdetidsperioder",
     finnFaktum = TotalTrygdetidGrunnlag::beregnetTrygdetidPerioder,
     finnFelt = { it }
 )
 
 val maksTrygdetid = definerKonstant<TotalTrygdetidGrunnlag, Int>(
-    gjelderFra = TRYGDETID_OMS_DATO,
+    gjelderFra = TRYGDETID_DATO,
     beskrivelse = "Full trygdetidsopptjening er 40 år",
     regelReferanse = RegelReferanse("REGEL-TOTAL-TRYGDETID-MAKS-ANTALL-ÅR"),
     verdi = 40
 )
 
+val dagerPrMaanedTrygdetid = definerKonstant<TotalTrygdetidGrunnlag, Int>(
+    gjelderFra = TRYGDETID_DATO,
+    beskrivelse = "En måned trygdetid tilsvarer 30 dager",
+    regelReferanse = RegelReferanse("REGEL-TOTAL-TRYGDETID-DAGER-PR-MND-TRYGDETID"),
+    verdi = 30
+)
+
 val totalTrygdetidFraPerioder = RegelMeta(
-    gjelderFra = TRYGDETID_OMS_DATO,
+    gjelderFra = TRYGDETID_DATO,
     beskrivelse = "Beregner trygdetid fra perioder",
     regelReferanse = RegelReferanse(id = "REGEL-TOTAL-TRYGDETID-SLÅ-SAMMEN-PERIODER")
-) benytter beregnetTrygdetidPerioder med { beregnetTrygdetidPerioder ->
-    beregnetTrygdetidPerioder.reduce { acc, period -> acc.plus(period) }.normalized()
+) benytter beregnetTrygdetidPerioder og dagerPrMaanedTrygdetid med { trygdetidPerioder, antallDagerEnMaanedTrygdetid ->
+    trygdetidPerioder
+        .reduce { acc, period -> acc.plus(period) }
+        .let {
+            val dagerResterende = it.days.mod(antallDagerEnMaanedTrygdetid)
+            val maanederOppjustert = it.months + (it.days - dagerResterende).div(antallDagerEnMaanedTrygdetid)
+            Period.of(it.years, maanederOppjustert, dagerResterende).normalized()
+        }
 }
 
 val totalTrygdetidAvrundet = RegelMeta(
-    gjelderFra = TRYGDETID_OMS_DATO,
-    beskrivelse = "Runder av trygdetid opp til hele år",
+    gjelderFra = TRYGDETID_DATO,
+    beskrivelse = "Avrunder trygdetid til nærmeste hele år",
```

Comment:
Ikke for å være vanskelig, men vil ikke det nærmeste året som blir valgt her ikke er det faktisk nærmeste i noen tilfeller? Mulig jeg overanalyserer her, men beskrivelsen og koden henger ikke helt sammen for meg